### ISSUE TITLE
DRY config for common config parameters (e.g. indent size, line length, indentation type)

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt#7698]
## Expected Behavior
Allow users to set common config parameters once instead of on multiple rules.

## Current Behavior
detekt-formatting's generated config file currently has 32 `indentSize` properties & 16 `maxLineLength` properties. We don't currently expose indentation style (tab, space) but that's used in many of ktlint's rules as well and should be introduced as its own common config parameter.

detekt core rules also have a couple of indent and line length settings.

## Context
Make it easy to configure common settings across rules.

**Comment by BraisGabin** [comment]
So we should support something like this:

```yaml
indentSize: 4
formatting:
  indentSize: 3
  maxLine:
    indentSize: 2
```

This should be doable with something like this:

```kotlin
    @Configuration("indentation size")
    private val indentSize by config(4, hierarchyLevel = -2)
```

This will tell the rule to find the `indentSize` configuration inside it or up to 2 levels bellow it. Other option:

```kotlin
    @Configuration("indentation size")
    private val indentSize by config(4, searchOnParents = true)
```

That will always search for the configuration on any of it's parents.

And even `searchOnParents = true` could be the default value.

The pro of the first option is that when we generate the default config we know where we should set the value of `indentSize`. Otherwise if we read:

```kotlin
    @Configuration("indentation size")
    private val indentSize by config(4, searchOnParents = true)
```

How should the default configuration be created:

```yaml
indentSize: 4
formatting:
  maxLine:
    active: true
```

or


```yaml
formatting:
  indentSize: 4
  maxLine:
    active: true
```

**Comment by cortinico** [comment]
> Make it easy to configure common settings across rules.

Folks could still use YAML anchors though to support this right?

**Comment by 3flex** [comment]
> use YAML anchors

They'd still need to leave in `indentSize` on every rule though which isn't very nice.

I don't think that allows for just setting something like this, where maxLine inherits the value from the rule set.:

```
formatting:
  indentSize: 3
  maxLine:
    active: true
```

With anchors wouldn't it be:
```
formatting:
  indentSize:  &indentSize 3
  maxLine:
    active: true
    indentSize: *indentSize
```

So `indentSize: *indentSize` still has to be added to every rule? It's much more concise if that's implicit (like cascading autoCorrect).

**Comment by cortinico** [comment]
> So `indentSize: *indentSize` still has to be added to every rule? It's much more concise if that's implicit (like cascading autoCorrect).

No strong opinion, but I think that here _Explicit is better than implicit_. 

As long as users don't have to repeat their own concrete values over and over, having all the config at the rule level will make it easier to reason about configs.

**Comment by 3flex** [comment]
This is a preference - we should support both. Some value explicit config, some value concise configs.

**Comment by 3flex** [comment]
Please see https://github.com/detekt/detekt/pull/9315#pullrequestreview-4262586064

I think we should merge that PR and I think we should also add the support for max line length and indent size and just set it on the parent configuration, not per rule.
